### PR TITLE
415 - text fields - manage text field style through theme configuration

### PR DESCRIPTION
**Pull request by paulinea**
By default Orange text fields are outlined.
The default text field style should be overridden to be filled in Innovation Cup theme.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -12,13 +12,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - \[Demo\] Add new entries in about screen ([#403](https://github.com/Orange-OpenSource/ods-android/issues/403))
 - \[Lib\] Add `OdsFloatingActionButton` and `OdsExtendedFloatingActionButton` components ([#109](https://github.com/Orange-OpenSource/ods-android/issues/109))
 - \[Lib\] Add `OdsListItem` composable signature with an `OdsListItemTrailing` as trailing parameter in order to manage accessibility ([#387](https://github.com/Orange-OpenSource/ods-android/issues/387))
+- \[ThemeConfigurationContract\] Add `textFieldStyle` boolean in the theme contract to allow to choose between outlined or filled text fields in a custom theme ([#415](https://github.com/Orange-OpenSource/ods-android/issues/415))
```
Comment (florentmaitre) [inline changelog.md:15]: Should we also note that `OdsComponentCustomizations` has been renamed to `OdsComponentsConfiguration` ?

File: changelog.md [hunk 2]
```diff
@@ -29,6 +31,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - \[Demo\] Screens in about section now use current theme colors ([#390](https://github.com/Orange-OpenSource/ods-android/issues/390))
 - \[Lib\] Fix cards vocalization when clickable ([#391](https://github.com/Orange-OpenSource/ods-android/issues/391))
 
+### Removed
+
+- \[Lib\] `OdsOutlinedTextfield` and `OdsPasswordOutlinedTextField` have been removed and replaced by `OdsTextField` and `OdsPasswordTextField`. Text fields appearance (outlined or filled) is now managed by the theme configuration. ([#415](https://github.com/Orange-OpenSource/ods-android/issues/415))
```
Comment (florentmaitre) [inline changelog.md:36]: I don't know if we need to keep two separate entries for this. Maybe we could just replace `[Lib]` by `[All]` and remove the one in `changed` section.

File: lib/src/main/java/com/orange/ods/compose/component/textfield/OdsTextField.kt [hunk 3]
```diff
@@ -115,84 +106,173 @@ fun OdsTextField(
...
+    onValueChange: (String) -> Unit,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    readOnly: Boolean = false,
+    label: String? = null,
+    placeholder: String? = null,
+    leadingIcon: Painter? = null,
+    leadingIconContentDescription: String? = null,
+    onLeadingIconClick: (() -> Unit)? = null,
+    trailing: @Composable (() -> Unit)? = null,
```
[the remark this answers is not in the document]
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/compose/component/textfield/OdsTextField.kt:196]: I think you're right cause in text fields the trailing element is always an icon. I'll rename it.

[the remark this answers is not in the document]
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/compose/component/textfield/OdsTextField.kt:196]: What do you think? The trailing element can also be a text.

**Review by florentmaitre — APPROVED** [review]

**Pull request merged**